static/app/views/issueList/groupSearchViewTabs/draggableTabBar.tsx

```diff
@@ -85,106 +97,230 @@ export function DraggableTabBar({
   onDiscardTempView,
   onSaveTempView,
 }: DraggableTabBarProps) {
-  const [selectedTabKey, setSelectedTabKey] = useState<Key>(tabs[0].key);
   // TODO: Extract this to a separate component encompassing Tab.Item in the future
   const [editingTabKey, setEditingTabKey] = useState<string | null>(null);
 
-  useEffect(() => {
-    if (!showTempTab && selectedTabKey === 'temporary-tab') {
-      setSelectedTabKey(tabs[0].key);
+  const {cursor: _cursor, page: _page, ...queryParams} = router.location?.query ?? {};
+  const selectedTabKey = queryParams.viewId ?? 'temporary-tab';
+  const setSelectedTabKey = useCallback(
+    (key: string) => {
+      router.replace({
+        query: {
+          ...queryParams,
+          viewId: key,
+        },
+        pathname: `/organizations/${orgSlug}/issues/`,
+      });
+    },
+    [orgSlug, queryParams, router]
+  );
+
+  const [tempTab, setTempTab] = useState<Tab | undefined>(
+    selectedTabKey === 'temporary-tab' && queryParams.query
+      ? {
+          id: 'temporary-tab',
+          key: 'temporary-tab',
+          label: 'Unsaved',
+          query: queryParams.query,
+          querySort: queryParams.sort,
+          queryCount: undefined,
+        }
+      : undefined
+  );
```

Comment:
@malwilley re your comment in the last PR: 

> I wonder if we can remove this state as well. We should be able to determine whether or not there is a temp tab by looking in the URL and seeing if there is no viewId, right?

Unfortunately I think we're going to need the tempTab state no matter what. If the user clicks off of the temp tab (but does not discard it), the temp tab is still shown in an un-selected state, and thus the component still needs to know what query, sort, etc. it has.

Reply:
Ah yes you're right! Guess we do need that extra state